### PR TITLE
Improve boot image generator

### PR DESCRIPTION
**Pull request by cahz**
* Support all zynq and zynqmp platforms available in TaPaSCo
* Linux version based on Vivado 2019.2
* Do not rely on pynq image anymore
* improved rootfs with ready-to-use environment for TaPaSCo

### CODE REVIEW

File: toolflow/boot/misc/resizefs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+sudo fdisk /dev/mmcblk0 <<EOF
```
Comment (jahofmann) [inline toolflow/boot/misc/resizefs:1]: Is this a stable API?

File: toolflow/boot/generate_boot_image.sh [hunk 2]
```diff
@@ -1,318 +1,327 @@
...
 JOBCOUNT=4
 SCRIPTDIR="$(dirname $(readlink -f $0))"
 DIR="$SCRIPTDIR/$BOARD/$VERSION"
 LOGDIR="$DIR/logs"
-CROSS_COMPILE=${CROSS_COMPILE:=arm-linux-gnueabihf-}
-ROOTFS_IMG="$SCRIPTDIR/rootfs.img"
-PYNQ_VERSION="pynq_z1_v2.1"
-PYNQ_IMAGE="$SCRIPTDIR/pynq/$PYNQ_VERSION.zip"
-PYNQ_IMAGE_URL="http://files.digilent.com/Products/PYNQ/$PYNQ_VERSION.img.zip"
+ROOTFS_URL="http://cdimage.ubuntu.com/ubuntu-base/releases/20.04/release/ubuntu-base-20.04-base-armhf.tar.gz"
```
Comment (jahofmann) [inline toolflow/boot/generate_boot_image.sh:10]: Worth using mirrors instead of direct URL?

File: toolflow/boot/generate_boot_image.sh [hunk 3]
```diff
@@ -1,318 +1,327 @@
...
-	else
-		echo "$DIR/BOOT.BIN already exists, skipping."
-	fi
-	if [[ ! -f $DIR/devicetree.dtb ]]; then
-		$DIR/linux-xlnx/scripts/dtc/dtc -I dts -O dtb -o $DIR/devicetree.dtb $SCRIPTDIR/pynq/devicetree.dts ||
-			return $(error_ret "$LINENO: could not build devicetree")
+fetch_arm_trusted_firmware() {
+	if [[ ! -d $DIR/arm-trusted-firmware ]]; then
+		echo "Fetching arm-trusted-firmware ..."
+		git clone --depth 1 https://github.com/Xilinx/arm-trusted-firmware.git $DIR/arm-trusted-firmware ||
```
Comment (jahofmann) [inline toolflow/boot/generate_boot_image.sh:148]: Move URLs up in Variables as done for the RootFS?

File: toolflow/boot/generate_boot_image.sh [hunk 4]
```diff
@@ -1,318 +1,327 @@
...
-		echo "$DIR/linux-xln already exists, skipping."
+		echo "$DIR/linux-xlnx already exists, skipping."
 	fi
 }
 
-fetch_u-boot () {
+fetch_u-boot() {
 	if [[ ! -d $DIR/u-boot-xlnx ]]; then
 		echo "Fetching u-boot $VERSION ..."
 		git clone -b xilinx-v$VERSION --depth 1 https://github.com/xilinx/u-boot-xlnx.git $DIR/u-boot-xlnx ||
```
Comment (jahofmann) [inline toolflow/boot/generate_boot_image.sh:138]: Move URLs up in Variables as done for the RootFS?

File: toolflow/boot/generate_boot_image.sh [hunk 5]
```diff
@@ -466,121 +573,164 @@ EOF
...
-		echo >&2 "$LINENO: WARNING: could not delete Jupyter stuff"
+	echo "Installing linux headers ..."
+	dusudo sh -c "cp -r $DIR/linux-xlnx $TO/usr/src/linux-headers-$(make -C $DIR/linux-xlnx kernelrelease -s)" ||
+		echo >&2 "$LINENO: WARNING: could not copy linux-xlnx"
+	echo "Preparing chroot environment"
+	dusudo sh -c "cp /usr/bin/qemu-arm-static $TO/usr/bin/"
+	dusudo sh -c "cp /usr/bin/qemu-aarch64-static $TO/usr/bin/"
+	dusudo sh -c "mount --bind /dev $TO/dev/"
+	dusudo sh -c "chroot $TO << EOF
+echo 'nameserver 8.8.4.4' | tee -a /etc/resolv.conf
```
Comment (jahofmann) [inline toolflow/boot/generate_boot_image.sh:646]: Secondary Nameserver?

File: toolflow/boot/misc/resizefs (hunk 1 above)
Comment (cahz) [reply toolflow/boot/misc/resizefs:1]: it is used by raspi-conf, so I assume it

File: toolflow/boot/generate_boot_image.sh (hunk 2 above)
Comment (cahz) [reply toolflow/boot/generate_boot_image.sh:10]: Should not matter for this file size (< 25 MB)

Comment (jahofmann) [reply toolflow/boot/generate_boot_image.sh:10]: Was more a question of: Is this link reliable, or would a mirrorlist or something increase reliability.

Comment (cahz) [reply toolflow/boot/generate_boot_image.sh:10]: The main ubuntu server should be reliable enough, would not spend any time on that.

**Pull request merged**